server/src/main/java/io/druid/metadata/SQLMetadataSupervisorManager.java

```diff
@@ -91,7 +92,7 @@ public Void withHandle(Handle handle) throws Exception
                     getSupervisorsTable()
                 )
             )
-                  .bind("id", String.format("%s_%s", id, version))
+                  .bind("id", String.format("%s_%s_%s", id, version, UUIDUtils.generateUuid()))
```

Comment:
@gianm cool, auto incrementing sounds good to me. It would only fix this bug if we made the version column the primary key as well (and might as well get rid of the id column), is that along the lines of what you were thinking?